From c3a4609d733423b064e573da1ad1de5de7208f1d Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Thu, 3 Mar 2016 17:19:30 -0800
Subject: [PATCH] laird: wb50: Move GPIO-3 to an input with pullup

Changing this supports having one input that we can utilize to wake from
suspend. We can't set the pullup in the device tree, so we have to set
on boot.

Bug: 9116
---
 driver/hw_init_hook.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/driver/hw_init_hook.c b/driver/hw_init_hook.c
index aaf876e..81420ba 100644
--- a/driver/hw_init_hook.c
+++ b/driver/hw_init_hook.c
@@ -29,7 +29,7 @@ void hw_init_hook(void)
 		 */
 		{"GPIO_5",	AT91C_PIN_PA(0),	0, PIO_DEFAULT, PIO_INPUT},  /* BOOT1, has pullup on BB40 */
 		{"GPIO_6",	AT91C_PIN_PA(3),	0, PIO_DEFAULT, PIO_INPUT},  /* BOOT2, has pullup on BB40 */
-		{"GPIO_3",	AT91C_PIN_PA(10),	0, PIO_DEFAULT, PIO_OUTPUT},  /* GPIO2 */
+		{"GPIO_3",	AT91C_PIN_PA(10),	0, PIO_PULLUP, PIO_INPUT},  /* GPIO2 */
 		{"LED0",	AT91C_PIN_PA(12),	0, PIO_DEFAULT, PIO_OUTPUT},
 		{"GPIO_4",	AT91C_PIN_PA(14),	0, PIO_DEFAULT, PIO_OUTPUT},  /* GPIO3 */
 		{"STAT0",	AT91C_PIN_PA(22),	0, PIO_DEFAULT, PIO_OUTPUT},
-- 
1.9.1

